test/augmentation/test_augmentation.py

```diff
@@ -24,6 +25,794 @@
+        test_transform = torch.rand(expected_transformation_shape, device=self.device, dtype=self.dtype)
+
+        output = augmentation(test_input, params=generated_params)
+        assert output.shape[0] == batch_shape[0]
+
+        output, transformation = augmentation(test_input, params=generated_params, return_transform=True)
+        assert output.shape[0] == batch_shape[0]
+        assert transformation.shape == expected_transformation_shape
+
+        output, final_transformation = augmentation(
+            (test_input, test_transform), params=generated_params, return_transform=True)
+        assert output.shape[0] == batch_shape[0]
+        assert final_transformation.shape == expected_transformation_shape
+        assert_allclose(final_transformation, transformation @ test_transform)
+
+        output, transformation = augmentation((test_input, test_transform), params=generated_params)
+        assert output.shape[0] == batch_shape[0]
+        assert transformation.shape == expected_transformation_shape
+        assert (transformation == test_transform).all()
+
+    def _test_smoke_return_transform_implementation(self, params):
+        batch_shape = (4, 3, 5, 6)
+        expected_transformation_shape = torch.Size((batch_shape[0], 3, 3))
+        test_input = torch.ones(batch_shape, device=self.device, dtype=self.dtype)
+        augmentation = self._create_augmentation_from_params(**params, return_transform=True)
+        generated_params = augmentation.generate_parameters(batch_shape)
+        test_transform = torch.rand(expected_transformation_shape, device=self.device, dtype=self.dtype)
+
+        output, transformation = augmentation(test_input, params=generated_params)
+        assert output.shape[0] == batch_shape[0]
+        assert transformation.shape == expected_transformation_shape
+
+        output, final_transformation = augmentation((test_input, test_transform), params=generated_params)
+        assert output.shape[0] == batch_shape[0]
+        assert final_transformation.shape == expected_transformation_shape
+        assert_allclose(final_transformation, transformation @ test_transform)
+
+        output, final_transformation = augmentation(
+            (test_input, test_transform), params=generated_params, return_transform=True)
+        assert output.shape[0] == batch_shape[0]
+        assert final_transformation.shape == expected_transformation_shape
+        assert_allclose(final_transformation, transformation @ test_transform)
+
+    def _test_cardinality_implementation(self, input_shape, expected_output_shape, params):
+
+        # p==0.0
+        augmentation = self._create_augmentation_from_params(**params, p=0.0)
+        test_input = torch.rand(input_shape, device=self.device, dtype=self.dtype)
+        output = augmentation(test_input)
+        assert len(output.shape) == 4
+        assert output.shape == torch.Size((1,) * (4 - len(input_shape)) + tuple(input_shape))
```

Comment:
what's this supposed to be doing when prob is zero? `output_shape==input.shape` no ?